### PR TITLE
5.11.0 migration

### PR DESCRIPTION
**Pull request by gomathyk**

### CODE REVIEW

File: en/docs/setup/changing-the-hostname.md [hunk 1]
```diff
@@ -3,14 +3,35 @@
 This section guides you through changing the hostname of WSO2 Identity
 Server.
 
-1.  Change the hostname ( Example: `is.dev.wso2.com`) in
-    ` <IS_HOME>/repository/conf/deployment.toml`.
+1.  Change the hostname 
+
+    Server hostname for internal API calls, is by default configured as `localhost`. This configuration is utilized 
+    to build the internal absolute url of a service endpoint, that will be consumed at places that will generate 
+    internal API calls. Hence, when configuring the hostname you can follow one from the below two options
```
Comment (sherenewso2) [inline en/docs/setup/changing-the-hostname.md:10]:
```suggestion
    to build the internal absolute url of a service endpoint that will be consumed whenever internal API calls are generated. To configure the hostname, follow one of the two options given below
```

File: en/docs/setup/changing-the-hostname.md [hunk 2]
```diff
@@ -3,14 +3,35 @@
...
+
+    Server hostname for internal API calls, is by default configured as `localhost`. This configuration is utilized 
+    to build the internal absolute url of a service endpoint, that will be consumed at places that will generate 
+    internal API calls. Hence, when configuring the hostname you can follow one from the below two options 
+    according to your requirement.
+
+    - **Option 1**
+
+    If you are configuring the hostnames in production for or a load-balanced or proxied deployment, you need 
+    to configure only the `hostname` as follows in `<IS_HOME>/repository/conf/deployment.toml`.
```
Comment (sherenewso2) [inline en/docs/setup/changing-the-hostname.md:16]:
```suggestion
    If you are configuring the hostnames in production for or a load-balanced or proxied deployment, configure only the `hostname` as follows in `<IS_HOME>/repository/conf/deployment.toml`.
```

File: en/docs/setup/migrating-userstore-managers.md [hunk 3]
```diff
@@ -0,0 +1,568 @@
...
+
+## Dry Run
+
+Dry-run capability in the user store migrator will allow the migration utility to validate the system for the 
+user store configurations and generate a report regarding the migration compatibility. In the generated migration report, 
+if there are any warnings in the generated migration report, it is recommended to contact the WSO2 support to identify the best migration strategy.
+
+### How to Run
+
+Configure the migration report path using the value of `reportPath` `<IS_HOME>/migration-resources/migration-config.yaml` in the migration-config.yaml.
```
Comment (sherenewso2) [inline en/docs/setup/migrating-userstore-managers.md:14]:
```suggestion
Configure the migration report path using the value of `reportPath` `<IS_HOME>/migration-resources/migration-config.yaml` in the `migration-config.yaml`. 
```

File: en/docs/setup/migrating-userstore-managers.md [hunk 4]
```diff
@@ -0,0 +1,568 @@
...
+## Migrating User Stores
+
+If you have multiple tenants and multiple user stores and you only need to migrate a few of them, refer to the 
+configuration section below. If you need to migrate all of them, use the `migrateAll` property (This is set to true by default).  
+
+### Special Scenarios
+
+If the user store is an LDAP user store and SCIM is enabled for that user store, migrating that user store is not
+ required. 
+As SCIM will create a user id for the users in that user store, SCIM id can be used as the unique user id. To do that,
```
Comment (sherenewso2) [inline en/docs/setup/migrating-userstore-managers.md:40]:
```suggestion
As SCIM will create a user ID for the users in that user store, the SCIM ID can be used as the unique user ID. To do that, 
```

File: en/docs/setup/migrating-userstore-managers.md [hunk 5]
```diff
@@ -0,0 +1,568 @@
...
+
+If you have multiple tenants and multiple user stores and you only need to migrate a few of them, refer to the 
+configuration section below. If you need to migrate all of them, use the `migrateAll` property (This is set to true by default).  
+
+### Special Scenarios
+
+If the user store is an LDAP user store and SCIM is enabled for that user store, migrating that user store is not
+ required. 
+As SCIM will create a user id for the users in that user store, SCIM id can be used as the unique user id. To do that, 
+change the `user_id_attribute` to the value of SCIM id, in the `<IS_HOME>/repository/conf/deployment.toml` file.
```
Comment (sherenewso2) [inline en/docs/setup/migrating-userstore-managers.md:41]:
```suggestion
change the `user_id_attribute` to the value of the SCIM ID, in the `<IS_HOME>/repository/conf/deployment.toml` file.
```

File: en/docs/setup/migrating-userstore-managers.md [hunk 6]
```diff
@@ -0,0 +1,568 @@
...
+
+## Configurations
+
+| Property Name | Description |
+| ------------- | ----------- |
+| migrateAll | Migrate all the tenants and all the user store domains in it. |
+| reportPath | Absolute path for the dry report. This is required in the dry run mode. |
+
+
+Following configurations are only needed to migrate a few tenants. Configuration format should be similar to the one mentioned below. tenant-param is a place holder name to represent each tenant uniquely. Ex: tenant1
```
Comment (sherenewso2) [inline en/docs/setup/migrating-userstore-managers.md:51]: Check formatting and grammar in this sentence and monospace parameter or placeholder names. The standard abbreviation for example is `e.g.,`

File: en/docs/setup/migrating-userstore-managers.md [hunk 7]
```diff
@@ -0,0 +1,568 @@
...
+  parameters:
+    <tenant-param>:
+```
+
+| Property Name | Description |
+| ------------- | ----------- |
+| tenantDomain | Domain name of the tenant. (Mandatory) |
+| increment | Number of users to be updated in each iteration. (Optional) |
+| startingPoint | Where should the migration start from (Offset). This is useful if the migration stopped middle and needs to restart. (Optional) |
+| scimEnabled | Whether SCIM enabled for user stores in this tenant. (Optional) |
```
Comment (sherenewso2) [inline en/docs/setup/migrating-userstore-managers.md:66]:
```suggestion
| scimEnabled | Whether SCIM is enabled for user stores in this tenant. (Optional) |
```

File: en/docs/setup/migrating-userstore-managers.md [hunk 8]
```diff
@@ -0,0 +1,568 @@
...
+  order: 7
+  parameters:
+    <tenant-param>:
+```
+
+| Property Name | Description |
+| ------------- | ----------- |
+| tenantDomain | Domain name of the tenant. (Mandatory) |
+| increment | Number of users to be updated in each iteration. (Optional) |
+| startingPoint | Where should the migration start from (Offset). This is useful if the migration stopped middle and needs to restart. (Optional) |
```
Comment (sherenewso2) [inline en/docs/setup/migrating-userstore-managers.md:65]:
```suggestion
| startingPoint | Where should the migration start from (Offset). This is useful if the migration stopped in the middle and needs to restart. (Optional) |
```

File: en/docs/setup/migrating-userstore-managers.md [hunk 9]
```diff
@@ -0,0 +1,568 @@
...
+| scimEnabled | Whether SCIM enabled for user stores in this tenant. (Optional) |
+| migratingDomains | List of comma separated domain names which should be migrated in this domain. (Optional) |
```
Comment (sherenewso2) [inline en/docs/setup/migrating-userstore-managers.md:67]:
```suggestion
| migratingDomains | List of comma-separated domain names that should be migrated to this domain. (Optional) |
```

File: en/docs/setup/migrating-userstore-managers.md [hunk 10]
```diff
@@ -0,0 +1,568 @@
...
+| forceUpdateUserId | Mark whether user IDs should be updated even though there is already an ID there. (Optional) |
```
Comment (sherenewso2) [inline en/docs/setup/migrating-userstore-managers.md:68]:
```suggestion
| forceUpdateUserId | Mark whether user IDs should be updated even though there is already an existing ID. (Optional) |
```

File: en/docs/setup/migrating-userstore-managers.md [hunk 11]
```diff
@@ -0,0 +1,568 @@
...
+
+## Database changes for migration
+
+These steps should be carried out for the old database before the migration. A backup of UM database should be taken and database triggers should be set to update the backup database based on the updates of the live database. After performing the following steps the backup database should be migrated to the next version.
+
+1.  If you have JDBC secondary user stores with SCIM disabled, execute the following queries on the um database. This will add a `UM_USER_ID` column to the `UM_USER` table with a random `UUID` as the default value for `UM_USER_ID`.
```
Comment (sherenewso2) [inline en/docs/setup/migrating-userstore-managers.md:74]: Here "um" is simple but in other places it is UM. let's keep it consistent

File: en/docs/setup/migrating-userstore-managers.md [hunk 12]
```diff
@@ -0,0 +1,568 @@
...
+| increment | Number of users to be updated in each iteration. (Optional) |
+| startingPoint | Where should the migration start from (Offset). This is useful if the migration stopped middle and needs to restart. (Optional) |
+| scimEnabled | Whether SCIM enabled for user stores in this tenant. (Optional) |
+| migratingDomains | List of comma separated domain names which should be migrated in this domain. (Optional) |
+| forceUpdateUserId | Mark whether user IDs should be updated even though there is already an ID there. (Optional) |
+
+## Database changes for migration
+
+These steps should be carried out for the old database before the migration. A backup of UM database should be taken and database triggers should be set to update the backup database based on the updates of the live database. After performing the following steps the backup database should be migrated to the next version.
+
```
Comment (sherenewso2) [inline en/docs/setup/migrating-userstore-managers.md:73]: is there a reason these are not given as numbered steps?

Comment (gomathyk) [reply en/docs/setup/migrating-userstore-managers.md:73]: It is given as a numbered list.

Comment (sherenewso2) [reply en/docs/setup/migrating-userstore-managers.md:73]: I mean can this be given as:

1. Take a backup of the UM database.
2. Set database triggers to update the backup database based on the updates of the live database. 
3. Migrate the backup database to the next version.

**Review by sherenewso2 — APPROVED** [review]

**Pull request merged**